testing/jormungandr-scenario-tests/src/scenario/controller.rs

```diff
@@ -88,38 +85,35 @@ impl ControllerBuilder {
         ControllerBuilder {
             title: title.to_owned(),
             controller_progress,
-            topology: None,
-            blockchain: None,
-            settings: None,
+            network_builder: Default::default(),
+            topology: Default::default(),
         }
     }
 
-    pub fn set_topology(&mut self, topology: Topology) {
+    pub fn topology(mut self, topology: Topology) -> Self {
         self.controller_progress.inc(1);
-        self.topology = Some(topology)
+        self.topology = topology.clone();
+        self.network_builder = self.network_builder.topology(topology);
+        self
     }
 
-    pub fn set_blockchain(&mut self, blockchain: Blockchain) {
+    pub fn blockchain(mut self, blockchain: Blockchain) -> Self {
         self.controller_progress.inc(1);
-        self.blockchain = Some(blockchain)
-    }
-
-    pub fn build_settings(&mut self, context: &mut ContextChaCha) {
-        self.controller_progress.inc(1);
-        let topology = self.topology.clone().expect("topology not set");
-        let blockchain = self.blockchain.clone().expect("blockchain not defined");
-        self.settings = Some(Settings::prepare(topology, blockchain, context));
-        self.controller_progress.inc(5);
+        self.network_builder = self.network_builder.blockchain_config(blockchain);
+        self
     }
 
     pub fn build(self, context: ContextChaCha) -> Result<Controller> {
+        self.controller_progress.inc(1);
```

Comment:
I don't understand what this progress bar is supposed to represent. We increment by 1, then clear it, then increment it by 5. And why do we increment it when we set the topology and blockchain?

Reply:
Yes, you are right. It doesn't make much of a sense. I've change a bit a mechanism in review fixes